File: src/main/java/org/ssv/model/QualityAttribute.java
package org.ssv.model;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import javax.persistence.*;

/**
 * Represents a quality attribute entity.
 */
@Entity
@Table(name = "quality_attribute")
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "type", discriminatorType = DiscriminatorType.STRING)
@Data
@NoArgsConstructor
public abstract class QualityAttribute implements Serializable {

    /**
     * The unique identifier for the quality attribute.
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private int id;

    /**
     * The name of the quality attribute.
     */
    @Column(name = "name")
    private String name;

    /**
     * The category of the quality attribute.
     */
    @Enumerated(EnumType.STRING)
    @Column(name = "category")
    private Category category;

    /**
     * The enum representing the category of the quality attribute.
     */
    public enum Category {
        SECURITY, PERFORMANCE_EFFICIENCY, MAINTAINABILITY
    }
}

File: src/main/app/src/interfaces/QualityAttribute.ts
export enum Category {
    SECURITY = "SECURITY",
    PERFORMANCE_EFFICIENCY = "PERFORMANCE_EFFICIENCY",
    MAINTAINABILITY = "MAINTAINABILITY"
}

export enum Relevance {
    HIGH = "HIGH",
    MEDIUM = "MEDIUM",
    LOW = "LOW",
    NONE = "NONE"
}

// Interface for QualityAttribute
export interface QualityAttribute {
    name: string;
    category: Category;
}
export interface QualityAttributeMS extends QualityAttribute {
    relevance: Relevance;
}

export interface QualityAttributeSR extends QualityAttribute {
    impactsPositively: boolean;
}